### ISSUE TITLE
[master issue] option to use custom IdP on browser extension

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-browser#5311]
Today, the browser extension uses Google authentication / IdP for two purposes:
1. authorizing use of Google APIs (gmail permissions) and then authenticating on those APIs
2. authenticating oneself when communicating with backend / Enterprise Server

For the second usecase, it would be safer to use some other IdP if the customer has that option. That way, Google cannot forge authentication tokens and steal the keys from EKM. Today, to prevent the potential key theft from EKM by Google, customers have to either run EKM behind a firewall on internal network, or not run it at all.

Allowing a custom IdP for authenticating especially with EKM would allow us to run EKM for our customers, meaning easier deployment.

This will mean, on such deployments when custom IdP is configured, the user will have to authenticate twice during setup. Once with Google for Gmail API and permissions, and once again with the custom IdP. (there will be two authentication popups in sequence)

The steps would be:
 - [x] #5316
 - [x] #5317
 - [x] #5318
 - [x] #5319
 - [x] #5444
 - [x] #5799
 - [x] #5801
 - [ ] #5929

**Comment by ioanmo226** [comment]
For the last task, I think I need @sosnovsky and @martgil  help.
I've tested this situation and it works well in my side though.

**Comment by ioanmo226** [comment]
Could you try to build extension from master branch and test this situation?

**Comment by sosnovsky** [comment]
Yes, I'll test such situation. Probably it's possible to implement such ui test?

**Comment by ioanmo226** [comment]
UI test (for such case) was already added in previous PR I think.
We set invalid customIdpIdToken and removed custom_idp_token_refresh(which is same case when custom domain authentication was added after extension was already set up)

> await BrowserRecipe.setInMemoryStore(settingsPage, acct, 'customIdpIdToken', MockJwt.new(acct, 100));
> await settingsPage.setLocalStorage(`cryptup_${emailKeyIndex(acct, 'custom_idp_token_refresh')}`, null);

**Comment by sosnovsky** [comment]
Ah, I see, yes, seems to be similar to the described situation, then no need for separate test.

**Comment by ioanmo226** [comment]
How's this test going?

**Comment by sosnovsky** [comment]
Haven't fully tested it yet, but seems to work well after custom IdP was added after previous non-custom configuration.
However, sometimes I get such popup on my personal @flowcrypt.com account, may be related to my test local configuration changes, as it's quite random and usually disappears after browser restart or extension reinstall:

<img width="1000" alt="Screenshot 2024-09-05 at 13 28 19" src="https://github.com/user-attachments/assets/23e0e37d-cff6-4d57-9805-9dc8748a6244">

**Comment by ioanmo226** [comment]
Let me check

**Comment by sosnovsky** [comment]
> Let me check

Probably it's related to my local test configuration, not sure if it'll occur for real users, so let's not try to fix it for now.

**Comment by ioanmo226** [comment]
I think this one is already finished?

**Comment by ioanmo226** [comment]
@sosnovsky ?

**Comment by sosnovsky** [comment]
I think we haven't fully tested process of enabling/disabling custom IdP configurations, but implementation seems to be complete. Filed separate issue for testing this functionality, so we won't forget about it.
